### ISSUE TITLE
error code 11050 for IPStatus not listed

### ISSUE THREAD

**Issue by rjrizzuto** [dotnet/runtime#101999]
### Description

11050 (general failure) is returned from a Ping.Send when wifi is turned off.  That code is not listed in the enums for the IPStatus class

### Reproduction Steps

Call Ping.Send with wifi is disabled.  Print the reply.status field.

### Expected behavior

GeneralFailure is printed

### Actual behavior

11050 is printed

### Configuration

.net 8
Windows 11
x64
Unsure if it is specific to this platform - can't test on Linux currently

**Comment by antonfirsov** [comment]
It would be nice to have an enum member to map [`IP_GENERAL_FAILURE=1105`](https://learn.microsoft.com/en-us/windows/win32/api/ipexport/ns-ipexport-icmp_echo_reply32) to, but I assume it would be platform specific. @wfurt are there other Windows-specific values in [`IPStatus`](https://learn.microsoft.com/en-us/dotnet/api/system.net.networkinformation.ipstatus)?

**Comment by rjrizzuto** [comment]
It seems like a catch all error code to report "other".  I imagine other platforms might need that as well.

**Comment by liveans** [comment]
Triage: Not critical for 9.0. Moving to future. Contributions are welcome. We'll take it into 9.0 if the time permits.

**Comment by piotreksda** [comment]
Hi, I am new here. I have looked into this a bit and was wondering if just adding `GeneralFailure = 11050` to the `enum IPStatus` is enough. I was also wondering if I should also change something in the `MapV4TypeToIPStatus` and `MapV6TypeToIPStatus` methods but here I would ask for help though.

**Comment by liveans** [comment]
> Hi, I am new here. I have looked into this a bit and was wondering if just adding `GeneralFailure = 11050` to the `enum IPStatus` is enough. I was also wondering if I should also change something in the `MapV4TypeToIPStatus` and `MapV6TypeToIPStatus` methods but here I would ask for help though.

Hello @piotreksda, this should be enough but while we were triaging this issue, we overlooked that `IPStatus` is public API, so this should go through `API Proposal` first, and only after then that we can implement this.

**Comment by anthonymastreanvae** [comment]
Looks like you're missing a few more, too:
* IP_BUF_TOO_SMALL 11001
* IP_BAD_REQ 11011
* IP_OPTION_TOO_BIG 11017

https://learn.microsoft.com/en-us/windows/win32/api/ipexport/ns-ipexport-icmp_echo_reply32

**Comment by anthonymastreanvae** [comment]
I'm also curious why the enum duplicates a value, which violates [CA1069: Enums should not have duplicate values](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1069):

* DestinationProhibited = 11004
* DestinationProtocolUnreachable = 11004

https://learn.microsoft.com/en-us/dotnet/api/system.net.networkinformation.ipstatus?view=net-9.0#fields 

The win32 structure does not even define "destination prohibited", only "destination protocol unreachable".

* IP_DEST_PROT_UNREACHABLE = 11004

https://learn.microsoft.com/en-us/windows/win32/api/ipexport/ns-ipexport-icmp_echo_reply32#members

The specification defines both.
 
* 2 = Protocol Unreachable
* 13 = Communication Administratively Prohibited

https://datatracker.ietf.org/doc/html/rfc1812#autoid-105

Furthermore, the DestinationProhibited name is correctly described (emphasis mine):

> The ICMPv6 echo request failed because contact with the destination computer is _administratively prohibited_.

But the DestinationProtocolUnreachable name is, I think, mistakenly described (emphasis mine):

> This value is described in [IETF RFC 1812](https://tools.ietf.org/html/rfc1812) as _Communication Administratively Prohibited_.
